module/trie.c

```diff
@@ -0,0 +1,271 @@
+/* This file implements a new module native data type called "TRIE".
+ * The data structure is based on the idea of a trie (aka a prefix 
+ * tree) and largely works the same way.
+ */
+#define IN_TRIE 1
+#define NOT_IN_TRIE 0 
+#define PARTIAL_IN_TRIE (-1)
+
+#include "../redismodule.h"
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <math.h>
+
+
+static RedisModuleType *trie;
+
+/* ========================== Internal data structure (Bare bones functions)  ======================= */
+
+struct trie {
+    char current; 
+        // The first trie_t will be '/0' for any Trie.
+    struct trie **children;
+         // ALPHABET_SIZE is 256 for all possible characters.
+    int is_word; 
+        // if is_word is 1, indicates that this is the end of a word. Otherwise 0.
+    struct trie *parent;
+        // parent trie for traversing backwards
+};
+
+/*
+    Creates and allocates memory for new trie.
+    
+    Parameters:
+     - current: A char for the current character
+    
+    Returns:
+     - A pointer to the trie, or REDISMODULE_ERR if a pointer 
+       cannot be allocated
+*/
+struct trie *trie_new (char current)
+{
+    struct trie *t = RedisModule_Calloc(1,sizeof(struct trie));
+    if (t == NULL){
+        fprintf(stderr, "Could not allocate memory for trie\n");
+        return NULL;
+    } 
+    t->current = current;
+    t->children = RedisModule_Calloc(256,sizeof(struct trie *));
+    if (t->children == NULL){
+        fprintf(stderr, "Could not allocate memory for t->children\n");
+        return NULL;
+    }
+    t->is_word = 0;
+    t->parent = NULL;
+    return t;
+}
+
+/*
+    Free an entire trie.
+    Parameters:
+     - t: A trie pointer
+    
+    Returns:
+     - Always returns 0
+*/
+int trie_free(struct trie *t)
+{
+    if (t != NULL) {
+        for (int i=0; i<256; i++ ){
```

Comment:
This line should be `for (int i = 0; i < 256; i++) {`. There are many more occurrences of whitespace issues throughout this file, please fix them all.